### PR TITLE
[Automation][Cephadm][tier2]Add matching subnets scenario

### PR DESCRIPTION
**Pull request by MohitBis**
[Automation][Cephadm][tier2]Add matching subnets scenario

### CODE REVIEW

**Comment by MohitBis** [comment]
Logs:

RHOSD: http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-D6TBNT/

RHOS01: http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-Q3FF1J/

**Review by pranavprakash20 — DISMISSED** [review]
LGTM

File: tests/cephadm/test_cephadm_change_public_network.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    config = kwargs.get("config")
+
+    # Get public network
+    public_network = config.get("public_network")
+
+    # Get installer node
+    installer = ceph_cluster.get_nodes(role="installer")[0]
+
+    # Set public network
+    CephAdm(installer).ceph.config.set(key="mon public_network", value=public_network)
```
Comment (vamahaja) [inline tests/cephadm/test_cephadm_change_public_network.py:18]: This should be -
```
CephAdm(installer).ceph.config.set(daemon="mon", key="public_network", value=public_network)
```

File: tests/cephadm/test_cephadm_change_public_network.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+    # Get cluster health
+    out = CephAdm(installer).ceph.health(detail=True)
```
Comment (vamahaja) [inline tests/cephadm/test_cephadm_change_public_network.py:21]: Only health check validation is sufficient? Don't need to check any config ?

**Review by vamahaja — CHANGES_REQUESTED** [review]

File: tests/cephadm/test_cephadm_change_public_network.py (hunk 1 above)
Comment (MohitBis) [reply tests/cephadm/test_cephadm_change_public_network.py]: Thanks @vamahaja 
Updated

File: tests/cephadm/test_cephadm_change_public_network.py (hunk 2 above)
Comment (MohitBis) [reply tests/cephadm/test_cephadm_change_public_network.py:21]: Hi @vamahaja 

It is sufficient

Same validation performed in polarion TC(CEPH-83575381)

**Review by tintumathew10 — COMMENTED** [review]
LGTM

**Review by vamahaja — APPROVED** [review]

**Pull request merged**